scripts/protoc_gateway.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+<< 'DOCS'
+  Generate gRPC-Gateway and OpenAPI specs from internal_api definitions.
+  This script uses the existing HTTP annotations in the proto file.
+DOCS
+
+# When DEBUG env is set - print output of the scripts.
+if [[ -e $DEBUG ]];
+then
+  set -x
+fi
+
+INTERNAL_API_REPO=git@github.com:renderedtext/internal_api.git
+INTERNAL_API_OUT=pkg/protos
+GATEWAY_OUT=pkg/protos
+OPENAPI_OUT=api/swagger
+MODULE_NAME=github.com/superplanehq/superplane
+MODULES=(${1//,/ })
+INTERNAL_API_BRANCH=${2:-master}
+INTERNAL_API_FOLDER=${3:-"/tmp/internal_api"}
+
+# Install required third-party proto files if not already present
+install_third_party_protos() {
+  THIRD_PARTY_DIR=$INTERNAL_API_FOLDER/include/google/api
+  if [ ! -d "$THIRD_PARTY_DIR" ]; then
+    echo "$(bold "Installing Google API proto files")"
+    mkdir -p $THIRD_PARTY_DIR
+    curl -L https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/annotations.proto > $THIRD_PARTY_DIR/annotations.proto
+    curl -L https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/http.proto > $THIRD_PARTY_DIR/http.proto
+    curl -L https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/field_behavior.proto > $THIRD_PARTY_DIR/field_behavior.proto
+  fi
+
+  OPENAPI_DIR=$INTERNAL_API_FOLDER/include/protoc-gen-openapiv2/options
+  if [ ! -d "$OPENAPI_DIR" ]; then
+    echo "$(bold "Installing OpenAPI proto files")"
+    mkdir -p $OPENAPI_DIR
+    curl -L https://raw.githubusercontent.com/grpc-ecosystem/grpc-gateway/main/protoc-gen-openapiv2/options/annotations.proto > $OPENAPI_DIR/annotations.proto
+    curl -L https://raw.githubusercontent.com/grpc-ecosystem/grpc-gateway/main/protoc-gen-openapiv2/options/openapiv2.proto > $OPENAPI_DIR/openapiv2.proto
+  fi
+}
+
+# Check if the proto file already has the necessary imports
+check_and_add_imports() {
```

Comment:
This will be made a lot simpler after https://github.com/superplanehq/superplane/pull/4 is merged, since then we can move the proto definition to this repo as well, and put all the annotations we need directly on it